matter_server/server/device_controller.py

```diff
@@ -1511,7 +1524,7 @@ def _on_mdns_operational_node_state(
             return
 
         # setup the node - this will (re) setup the subscriptions etc.
-        asyncio.create_task(self._setup_node(node_id))
+        asyncio.create_task(self._setup_node_with_retry(node_id))
```

Comment:
Let's store a reference to this task in the device controller. Make self._nodes_in_setup a dict that stores the node id as key and the task as value - that way we can also safely shutdown all these setup tasks at shutdown

Reply:
We do use call setup node from other places too, without create_task. So this needs a bit more refactoring :thinking:

Reply:
yes but now you are suggesting a long(er) running task

Reply:
I am creating a task always now, so we can cancel any running setup at the end.